substrate/frame/revive/src/benchmarking.rs

```diff
@@ -2330,6 +2336,283 @@ mod benchmarks {
+			hex!("5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133");
+
+		let signer_key = SigningKey::from_bytes(&signer_priv_key.into()).expect("valid key");
+
+		let signer_address = H160::from_slice(&signer_account_id);
+		let signer_caller = T::AddressMapper::to_fallback_account_id(&signer_address);
+
+		(signer_caller, signer_key, signer_address)
+	}
+
+	// Helper function to extract contract events from system events
+	fn get_contract_events<T: Config>() -> Vec<
+		frame_system::EventRecord<
+			<T as frame_system::Config>::RuntimeEvent,
+			<T as frame_system::Config>::Hash,
+		>,
+	> {
+		frame_system::Pallet::<T>::events()
+			.into_iter()
+			.filter(|event_record| {
+				// Use string representation to identify contract events
+				format!("{:?}", event_record.event).contains("ContractEmitted")
+			})
+			.collect()
+	}
+
+	// Helper function to create and sign a transaction for finalize_block benchmark
+	fn create_signed_transaction(
+		signer_key: &SigningKey,
+		target_address: H160,
+		value: U256,
+	) -> Vec<u8> {
+		let unsigned_tx: TransactionUnsigned = TransactionLegacyUnsigned {
+			to: Some(target_address),
+			value,
+			chain_id: Some(1.into()),
+			input: vec![].into(),
+			..Default::default()
+		}
+		.into();
+
+		let hashed_payload = sp_io::hashing::keccak_256(&unsigned_tx.unsigned_payload());
+		let (signature, recovery_id) =
+			signer_key.sign_prehash_recoverable(&hashed_payload).expect("signing success");
+
+		let mut sig_bytes = [0u8; 65];
+		sig_bytes[..64].copy_from_slice(&signature.to_bytes());
+		sig_bytes[64] = recovery_id.to_byte();
+
+		let signed_tx = unsigned_tx.with_signature(sig_bytes);
+
+		signed_tx.signed_payload()
+	}
+
+	// Macro to generate common finalize_block benchmark setup
+	macro_rules! setup_finalize_block_benchmark {
+		() => {{
+			// Setup test signer
+			let (signer_caller, signer_key, _signer_address) = create_test_signer::<T>();
+			whitelist_account!(signer_caller);
+
+			// Setup contract instance
+			let instance =
+				Contract::<T>::with_caller(signer_caller.clone(), VmBinaryModule::dummy(), vec![])?;
+			let origin = RawOrigin::Signed(signer_caller.clone());
+			let storage_deposit = default_deposit_limit::<T>();
+			let value = Pallet::<T>::min_balance();
+			let evm_value =
+				Pallet::<T>::convert_native_to_evm(BalanceWithDust::new_unchecked::<T>(value, 0));
+
+			// Create signed transaction
+			let signed_payload =
+				create_signed_transaction(&signer_key, instance.address, value.into());
+
+			// Setup block
+			let current_block = BlockNumberFor::<T>::from(1u32);
+			frame_system::Pallet::<T>::set_block_number(current_block);
+
+			(
+				signer_caller,
+				instance,
+				origin,
+				storage_deposit,
+				evm_value,
+				signed_payload,
+				current_block,
+			)
+		}};
+	}
+
+	// `c`: number of transactions to fit in the block
+	#[benchmark(pov_mode = Measured)]
+	fn finalize_block_transaction_processing(c: Linear<0, 100>) -> Result<(), BenchmarkError> {
```

Comment:
at first glance I am wondering why you have 2 benchmark with a single parameter, rather than one with the two paramters

this would generate a weight function similar to  
finalize_block_processing(events, transaction)

Then you can deduct:
- base cost          finalize_block_processing(0, 0)
- cost of 1 event  finalize_block_processing(1, 0) - finalize_block_processing(0, 0)
- cost of 1 tx        finalize_block_processing(0, 1) - finalize_block_processing(0, 0)

Reply:
Merged `finalize_block_transaction_processing(transaction_cnt)` and `finalize_block_event_processing(event_cnt)` into single  `finalize_block_processing(transaction_cnt, event_cnt)`.

Please keep in mind that there is yet another benchmark `finalize_block_event_data_processing(data_len)`. I left it as a separate one, for the sake of simplicity. But if you want I can merge it with the upper one as well.